### PR TITLE
Seekable transcodes

### PR DESCRIPTION
**Pull request by randomnicode**
Same PR as https://github.com/airsonic/airsonic/pull/1123

### CODE REVIEW

**Comment by eharris** [comment]
Ah thanks for pulling this in.  I was just about to work on submitting it but saw you had already done it.

File: airsonic-main/src/main/java/org/airsonic/player/controller/StreamController.java [hunk 1]
```diff
@@ -230,25 +232,37 @@ public void handleRequest(HttpServletRequest request, HttpServletResponse respon
...
                         if (n == -1) {
                             if (isPodcast || isSingleFile) {
+                                // Pad the output if needed to avoid content length errors on transcodes
+                                if (fileLengthExpected != null && bytesWritten < fileLengthExpected) {
+                                    sendDummy(buf, out, fileLengthExpected - bytesWritten);
+                                }
                                 break;
                             } else {
-                                sendDummy(buf, out);
+                                sendDummyDelayed(buf, out);
```
Comment (randomnicode) [inline airsonic-main/src/main/java/org/airsonic/player/controller/StreamController.java]: out of curiosity, why is this sent after a delay?

Comment (eharris) [reply airsonic-main/src/main/java/org/airsonic/player/controller/StreamController.java]: I just preserved the original behavior, renaming the function so its name was a better indication of what it does.  I surmise it is attempting to keep the stream alive while waiting for some other event to occur, but I'm not really sure under what circumstances it was used.

Comment (randomnicode) [reply airsonic-main/src/main/java/org/airsonic/player/controller/StreamController.java]: okay, i'll try to dig in what's happening

**Pull request merged**